b1ddi/resource_ipam_subnet_test.go

```diff
@@ -576,3 +576,53 @@ func testAccReadDhcpHost(t *testing.T) string {
 
 	return internalSecondary
 }
+
+func TestAccResourceSubnet_NextAvailableSubnetInAddressBlock(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+					resource "b1ddi_ip_space" "tf_acc_test_space" {
+  						name = "tf_acc_test_space"
+  						comment = "This IP Space is created by terraform provider acceptance test"
+					}
+					resource "b1ddi_address_block" "tf_acc_test_address_block" {
+						address = "192.168.1.0"
+  						name = "tf_acc_test_address_block"
+						cidr = 24
+						space = b1ddi_ip_space.tf_acc_test_space.id 
+  						comment = "This Address Block is created by terraform provider acceptance test"
+						tags = {
+							TestType = "Acceptance"
+						}
+					}
+					data "b1ddi_dhcp_hosts" "dhcp_host" {
+						filters = {
+							"name" = "%s"
+						}
+					}
+					resource "b1ddi_subnet" "tf_acc_test_subnet_nas" {
+						address = b1ddi_address_block.tf_acc_test_address_block.id
+						cidr = 25
+						comment = "This Subnet is updated by terraform provider acceptance test"
+						dhcp_host = data.b1ddi_dhcp_hosts.dhcp_host.results.0.id
+						name = "tf_acc_test_subnet"
+						space = b1ddi_ip_space.tf_acc_test_space.id	
+						lifecycle {
+    						ignore_changes = all
+  						}
+					}`, testAccReadDhcpHost(t)),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testCheckSubnetExists("b1ddi_subnet.tf_acc_test_subnet_nas"),
+
+					resource.TestCheckResourceAttr("b1ddi_subnet.tf_acc_test_subnet_nas", "cidr", "25"),
```

Comment:
Shouldn't we validate the host as well?

Reply:
Yes, we can. But the value stored wouldn't be the hostname but instead the ID. We will need to implement a function to get that from the hostname. Hence I decided to skip it.

Reply:
Updated the tests to check if this `dhcp_host` is set